app/models/casebook/concerns/contents.rb

```diff
@@ -0,0 +1,53 @@
+module Casebook::Concerns::Contents
+  extend ActiveSupport::Concern
+
+  def children
+    contents.where(['array_length(casebooks.ordinals, 1) = ?', ordinals.length + 1])
```

Comment:
Just to be more explicit: The major benefit of doing it this way is that we can pull out all the contents for a given book, in sorted order by table-of-contents number, in a single SQL query: `where(root_id: book.id).order(:ordinals)`. Pulling out children and subtrees is also easy by looking for ordinals having a given prefix. We don't need any complex SQL or multiple database hits to follow a parent-child tree. That list can then be more or less rendered straight to HTML (albeit needing the `tree` transformation for the `.chapter-contents` tags. That could be eliminated with slightly more complex HTML generation, though.) 

It adds complexity to inserting rows, since siblings need to be pushed over and all children in a subtree need to be updated. But those steps can also be done in single SQL queries (see the code in Casebook::Contents), and in trade we remove a bunch of complexity from the `SELECT` side, which I think is a good deal.